From 3fe1575937ece06bae29c98daafb65f986b70ef2 Mon Sep 17 00:00:00 2001
From: Quan Zhang <spring.zhang@nxp.com>
Date: Tue, 6 Sep 2016 03:34:24 +0800
Subject: [PATCH 1159/5242] MLK-13206 dcp: mx6sl: add missing components in
 dts

commit  aa26b06044d800651c520e97c1fa2fa24c3c606a from
https://source.codeaurora.org/external/imx/linux-imx.git

After MX6ULL DCP issue is fixed in commit 7a1cc1f, it introduces a new issue,
MX6SL will meet issue as no dcp clock is defined when initializing:
[    3.061344] mxs-dcp 20fc000.dcp: can't identify DCP clk: -2

On mx6sl, dcp clock is always on, so the patch use dummy as dcp clock directly.

Signed-off-by: Quan Zhang <spring.zhang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../devicetree/bindings/crypto/fsl-dcp.txt         |    2 +-
 arch/arm/boot/dts/imx6sl.dtsi                      |    4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/Documentation/devicetree/bindings/crypto/fsl-dcp.txt b/Documentation/devicetree/bindings/crypto/fsl-dcp.txt
index 76a0b4e..9c2ef99 100644
--- a/Documentation/devicetree/bindings/crypto/fsl-dcp.txt
+++ b/Documentation/devicetree/bindings/crypto/fsl-dcp.txt
@@ -10,7 +10,7 @@ Required properties:
 Example:
 
 dcp@80028000 {
-	compatible = "fsl,imx28-dcp", "fsl,imx23-dcp";
+	compatible = "fsl,imx6sl-dcp", "fsl,imx28-dcp", "fsl,imx23-dcp";
 	reg = <0x80028000 0x2000>;
 	interrupts = <52 53>;
 };
diff --git a/arch/arm/boot/dts/imx6sl.dtsi b/arch/arm/boot/dts/imx6sl.dtsi
index 73114fc..422913b 100644
--- a/arch/arm/boot/dts/imx6sl.dtsi
+++ b/arch/arm/boot/dts/imx6sl.dtsi
@@ -830,6 +830,10 @@
 				interrupts = <0 99 IRQ_TYPE_LEVEL_HIGH>,
 					     <0 100 IRQ_TYPE_LEVEL_HIGH>,
 					     <0 101 IRQ_TYPE_LEVEL_HIGH>;
+				/* DCP clock always on */
+				clocks = <&clks IMX6SL_CLK_DUMMY>;
+				clock-names = "dcp";
+				status = "okay";
 			};
 		};
 
-- 
1.7.9.5

